### PR TITLE
RHINENG-9528: Improve image build process

### PR DESCRIPTION
**Pull request by rexwhite**
Update the way we build the image for remediations.

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -3,42 +3,48 @@
...
-ENV APP_ROOT=/opt/app-root/src
+ENV APP_ROOT=/opt/app-root
 WORKDIR $APP_ROOT
 RUN mkdir -p $APP_ROOT/.npm/{_logs,_cacache} && chgrp -R 0 $APP_ROOT && chmod -R ug+rwX $APP_ROOT
 
-RUN npm install -g npm@10.5.1
+RUN npm install -g npm@10.5.2
 
-#--------------------- packages ---------------------
+RUN useradd -r -u 1001 -g 0 -m -d $APP_ROOT app_user
```
[the remark this answers is not in the document]
Comment (rexwhite) [reply build/Dockerfile:15]: The newer version of npm, required by the update to a newer version of node.js, which is required by RHINENG-9153, requires that a user have a home directory (which user 1001 does not) and the guidance from Red Hat's "how to use ubi images" is to create the user.

[the remark this answers is not in the document]
Comment (vkrizan) [reply build/Dockerfile:15]: Home directory can be set by `HOME` env var. We used that within Drift for poetry: https://github.com/RedHatInsights/drift-backend/blob/45433a19bfd725a28284cd6193efb7fd14d5cf68/Dockerfile#L52

[the remark this answers is not in the document]
Comment (rexwhite) [reply build/Dockerfile:15]: Oooh... THAT actually works and I like it much better!  I'll update the Dockerfile.  Thanks!

[the remark this answers is not in the document]
Comment (vkrizan) [reply build/Dockerfile:15]: I'm glad it works with npm as well. Cheers.

**Pull request merged**